src/librustc_codegen_llvm/mir/place.rs

```diff
@@ -174,7 +174,10 @@ impl PlaceRef<'ll, 'tcx> {
         let cx = bx.cx;
         let field = self.layout.field(cx, ix);
         let offset = self.layout.fields.offset(ix);
-        let align = self.align.min(self.layout.align).min(field.align);
+        let effective_field_align = self.align
+            .min(self.layout.align)
+            .min(field.align)
```

Comment:
It might help #54028 to remove this line, and rely *strictly* on the alignment of the type, combined with the best-case alignment for the offset. That way, we can hint LLVM about alignments that are higher than it can figure out by itself. Do we want to do that and do we expect a performance improvement?